drivers/gles3/shader_gles3.cpp

```diff
@@ -716,7 +716,7 @@ void ShaderGLES3::_initialize_version(Version *p_version) {
 
 void ShaderGLES3::version_set_code(RID p_version, const HashMap<String, String> &p_code, const String &p_uniforms, const String &p_vertex_globals, const String &p_fragment_globals, const Vector<String> &p_custom_defines, const LocalVector<ShaderGLES3::TextureUniformData> &p_texture_uniforms, bool p_initialize) {
 	Version *version = version_owner.get_or_null(p_version);
-	ERR_FAIL_NULL(version);
+	ERR_FAIL_NULL_MSG(version, "Shader compilation failed (most likely due to a syntax error). Shader code has been printed above for troubleshooting.");
```

Comment:
I don't think this is correct. I can't find any code path where this would be reached with a null `version` due to failing to compile. If compilation fails, the engine won't attempt to set the code on the version. 

The error in the bug report comes from `version_get_shader()` https://github.com/godotengine/godot/blob/06fbc8395b3c0ec6fa38588caea2ee94837f7b97/servers/rendering/renderer_rd/shader_rd.h#L166

It happens when you try to use a shader that previously failed compilation

Reply:
I've changed the message accordingly. Not sure if we can make it much more specific than what I put:

> Can't use a shader that previously failed compilation.